status/boost_check_library.py

```diff
@@ -83,6 +107,32 @@ def check_organization_include(self):
                 ''',
                 'org-inc-one')
     
+    def check_organization_macros(self):
+        libdir = os.path.join(self.library_dir,'include','boost',self.library_name)
+        if os.path.isdir(libdir):
+            badmacros = {}
+            for root,dirs,files in os.walk(libdir):
+                if (os.path.sep + 'detail' + os.path.sep) in root or root.endswith(os.path.sep + 'detail'):
```

Comment:
Better:
`if 'detail' in dirs: dirs.remove('detail')`

This avoids going down into the detail headers completely speeding up the traversal

Reply:
I don't think what you propose does that. We never use dirs in fact.

Reply:
It does: https://docs.python.org/3/library/os.html#os.walk

Reply:
> When topdown is True, the caller can modify the dirnames list in-place (perhaps using del or slice assignment), and walk() will only recurse into the subdirectories whose names remain in dirnames; this can be used to prune the search,

Reply:
Well, you learn something every day. Thanks.